tests/run.sh

```diff
@@ -169,9 +169,10 @@ for f in $tests; do
     # Tail the log and quit, when awesome quits.
     tail -n 100000 -f --pid $awesome_pid $awesome_log
 
-    if grep -q -E '^Error|assertion failed' $awesome_log; then
+    if ! grep -q -E '^Test finished successfully$' $awesome_log ||
+            grep -q -E '[Ee]rror|assertion failed' $awesome_log; then
```

Comment:
Whoops. Yeah, that's intentional, but I forgot the one below. Better be too greedy when looking for error messages (I noticed that the output of `lua -e 'error("")'` wouldn't be matched. This would now match the function name "error".)